src/context/Filter.context.tsx

```diff
@@ -0,0 +1,121 @@
+import { createContext, useState } from 'react';
+
+import { useProductsDataContext } from '@/hooks/useProductsDataContext';
+import { SortByEnum } from '@/interfaces/Filters';
+import type { Product } from '@/interfaces/Product';
+
+interface StateInterface {
+    filters: string[];
+    sortBy: SortByEnum;
+    search: string;
+}
+
+interface FilterContextInterface {
+    filteredProductsData: Product[];
+    filtersState: StateInterface;
+    toggleButtonFilter: (filter: string) => void;
+    handleSearcFilter: (string: string) => void;
+    handleSelectFilter: (string: SortByEnum) => void;
+}
+
+export const FilterContext = createContext<FilterContextInterface>({
+    filteredProductsData: [],
+    filtersState: {
+        filters: [],
+        sortBy: SortByEnum.HIGH_TO_LOW,
+        search: '',
+    },
+    toggleButtonFilter: () => {},
+    handleSearcFilter: () => {},
+    handleSelectFilter: () => {},
+});
+
+interface FilterContextProviderProps {
+    children: React.ReactNode;
+}
+
+function FilterContextProvider({ children }: FilterContextProviderProps) {
+    const { productsData } = useProductsDataContext();
+    const [filtersState, setFiltersState] = useState<StateInterface>({
+        filters: [],
+        sortBy: SortByEnum.HIGH_TO_LOW,
+        search: '',
+    });
+
+    function toggleButtonFilter(filter: string) {
+        setFiltersState((previousState) => {
+            if (previousState.filters.includes(filter)) {
+                const updatedState = structuredClone(previousState);
+                updatedState.filters.splice(updatedState.filters.indexOf(filter), 1);
+                return updatedState;
+            } else {
+                return {
+                    ...previousState,
+                    filters: [...previousState.filters, filter],
+                };
+            }
+        });
+    }
+
+    function handleSearcFilter(string: string) {
+        setFiltersState((previousState) => ({
+            ...previousState,
+            search: string,
+        }));
+    }
+
+    function handleSelectFilter(filter: SortByEnum) {
+        setFiltersState((previousState) => ({
+            ...previousState,
+            sortBy: filter,
+        }));
+    }
+
+    function filterData(dataArray: Product[]): Product[] {
```

Comment:
bad naming, 
- dont understand what data it is filter
- it is not filter data, this function make two things, filter something AND sort the result of filtering
So u collapse one of the MAIN programming principle
**Single Responsibility**
So please:
- split the function into two smaller and rename tham

```ts
const contextValue = {
        filtersState,
        filteredProductsData: filterProducts(sortProducts(productsData)),
        toggleButtonFilter,
        handleSearcFilter,
        handleSelectFilter,
    };
```

https://en.wikipedia.org/wiki/Single-responsibility_principle